collection.c

```diff
@@ -2539,21 +2558,19 @@ void php_mongodb_aggregate(zval *pipeline, zval *options, mongo_db *db, mongo_co
 	zval_add_ref(&collection->name);
 	zval_add_ref(&pipeline);
 
-
-	original_rp = collection->read_pref.type;
-	zend_hash_internal_pointer_reset(HASH_OF(pipeline));
-	while (zend_hash_get_current_data(HASH_OF(pipeline), (void **)&op) == SUCCESS) {
-		if (zend_symtable_exists(Z_ARRVAL_PP(op), "$out", strlen("$out") + 1)) {
-			if (collection->read_pref.type > MONGO_RP_PRIMARY_PREFERRED) {
-				mongo_manager_log(MonGlo(manager), MLOG_RS, MLOG_WARN, "Forcing aggregate with $out to run on primary");
-				php_error_docref(NULL TSRMLS_CC, E_WARNING, "Forcing aggregate with $out to run on primary");
-				collection->read_pref.type = MONGO_RP_PRIMARY;
-				break;
-			}
-		}
-		zend_hash_move_forward(HASH_OF(pipeline));
+	if (
+		collection->read_pref.type > MONGO_RP_PRIMARY &&
```

Comment:
Although this works, I think using != instead of > makes it more readable. Originally, the > made more sense as it would otherwise have 3 (or 4) equality comparisons.